File: StudioMedico/src/main/java/com/StudioMedico/entity/Prenotazione.java
package com.StudioMedico.entity;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Data
public class Prenotazione implements Serializable {


    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id")
    private int id;

    @Column(name = "data_prenotazione")
    private LocalDate dataPrenotazione;

    @Column(name = "ora_prenotazione")
    private LocalDateTime oraPrenotazione;

    @Column(name = "status_prenotazione")
    private String statusPrenotazione;

    @Column(name = "note")
    private String note;

    @ManyToOne
    @JoinColumn(name="id_paziente")
    private Paziente paziente;

    @ManyToOne
    @JoinColumn(name="id_medico")
    private Medico medico;

}

File: StudioMedico/StudioFE/src/app/models/Prenotazione.ts
import { Medico } from "./Medico";
import { Paziente } from "./Paziente";

export class Prenotazione{
    public id?: number;
    public dataPrenotazione?: Date;
    public oraPrenotazione?: Date;
    public statusPrenotazione?: string;
    public note?: string;
    public paziente?: Paziente;
    public medico?: Medico;
}